Data/HashSet.hs

```diff
@@ -4,25 +4,68 @@
 #endif
 
 ------------------------------------------------------------------------
--- |
--- Module      :  Data.HashSet
--- Copyright   :  2011 Bryan O'Sullivan
--- License     :  BSD-style
--- Maintainer  :  johan.tibell@gmail.com
--- Stability   :  provisional
--- Portability :  portable
---
--- A set of /hashable/ values.  A set cannot contain duplicate items.
--- A 'HashSet' makes no guarantees as to the order of its elements.
---
--- The implementation is based on /hash array mapped trie/.  A
--- 'HashSet' is often faster than other tree-based set types,
--- especially when value comparison is expensive, as in the case of
--- strings.
---
--- Many operations have a average-case complexity of /O(log n)/.  The
--- implementation uses a large base (i.e. 16) so in practice these
--- operations are constant time.
+{-|
+Module      :  Data.HashSet
+Copyright   :  2011 Bryan O'Sullivan
+License     :  BSD-style
+Maintainer  :  johan.tibell@gmail.com
+Stability   :  provisional
+Portability :  portable
+
+= Introduction
+
+'HashSet' allows you to store /unique/ elements, providing efficient
+insertion, lookups, and deletion. If you are storing sets of @Int@s consider
+using the @Data.IntSet@ from the
+<https://hackage.haskell.org/packages/containers containers> package.
+A 'HashSet' makes no guarantees as to the order of its elements.
+
+== Immutability
+
+@HashSet@s are /immutable/ which means that any update functions do not modify
+the set you passed in, they create a new set. In order to keep the changes
+you need to assign it to a new variable.
```

Comment:
Is this really necessary? We do assume that the reader has used Haskell before, right?

Reply:
Eh, a lot of the time when you're first getting into a languages you'll reach for data structure libraries to play around simple problems, so from my perspective it doesn't hurt and is a concrete example for newcomers. I could go either way though.

Reply:
Yeah. I'm slightly worried about losing the reader with obvious things like that. So I'm still slightly in favour of removing this.